src/org/httpkit/client.clj

```diff
@@ -193,7 +195,9 @@
                                 ;; only the 4 support now
                                 (case as :auto 1 :text 2 :stream 3 :byte-array 4))
         effective-proxy-url (if proxy-host (str proxy-host ":" proxy-port) proxy-url)
-        cfg (RequestConfig. method headers body timeout keepalive effective-proxy-url tunnel?)]
+        conn-timeout (if timeout timeout conn-timeout)
```

Comment:
Another interesting thing is that if you provide only a general timeout value T, then set connection and read timeouts to this value, you will actually get potentially 2 * T timeout. So defining it the other way around could be confusing.